tests/util.py

```diff
@@ -14,8 +14,11 @@ def clean_xml(xml_str):
     Remove whitespace from XML.
     """
     parser = etree.XMLParser(remove_blank_text=True)
-    return etree.tostring(
-        etree.XML(xml_str, parser=parser))
+    try:
+        return etree.tostring(
+            etree.XML(xml_str, parser=parser)).decode('utf-8')
+    except UnicodeError:
```

Comment:
updated.

On Tue, Oct 20, 2015 at 11:25 AM, George Schneeloch <
notifications@github.com> wrote:

> In tests/util.py
> <https://github.com/mitodl/xbundle/pull/48#discussion_r42534074>:
>
> > @@ -14,8 +14,11 @@ def clean_xml(xml_str):
> >      Remove whitespace from XML.
> >      """
> >      parser = etree.XMLParser(remove_blank_text=True)
> > -    return etree.tostring(
> > -        etree.XML(xml_str, parser=parser))
> > +    try:
> > +        return etree.tostring(
> > +            etree.XML(xml_str, parser=parser)).decode('utf-8')
> > +    except UnicodeError:
>
> I mentioned the AttributeError since it would happen if you tried to
> invoke decode on a Python 3 string object so you could use it to tell
> that it was already a unicode string. I'd prefer to let the UnicodeError
> fail the test since it may indicate something wrong, and the empty string
> returned may equal another empty string returned by another failing (but
> unequal) piece of XML.
>
> Sorry if I'm coming across as overly picky, just trying to be thorough. I
> think overall this is a good piece of work done very quickly and I
> appreciate your attention to detail.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mitodl/xbundle/pull/48/files#r42534074>.
>